tensorflow/contrib/periodic_resample/kernels/periodic_resample_op.h

```diff
@@ -211,7 +391,36 @@ void create_output_tensor(
       fill_periodic_tensor<tensorflow::int64>(context, desired_shape,
                                               input_tensor);
       break;
-    default:;
+    default:
+      ;
+  }
+}
+
+void create_grad_tensor(tensorflow::OpKernelContext* context,
+                        const tensorflow::Tensor& grad_tensor,
+                        const tensorflow::DataType& grad_tensor_type,
+                        const tensorflow::TensorShape& original_shape,
+                        const tensorflow::PartialTensorShape& desired_shape) {
+  // obligatory type switch
+  switch (grad_tensor_type) {
+    case tensorflow::DataTypeToEnum<float>::value:
```

Comment:
Consider using a macro to avoid duplicating the details of all instantiations here.

Reply:
Done